pkg/labels/selector.go

```diff
@@ -800,6 +801,14 @@ func SelectorFromSetParse(ls Set) (Selector, error) {
 // ParseSelector takes a string representing a selector and returns an
 // object suitable for matching, or an error.
 func ParseSelector(selector string) (Selector, error) {
+	return parseVersionedSelector(selector, "", "")
+}
+
+func ParseVersionedFieldSelector(selector, version, apiResource string) (Selector, error) {
+	return parseVersionedSelector(selector, version, apiResource)
+}
+
+func parseVersionedSelector(selector, version, apiResource string) (Selector, error) {
```

Comment:
Yes. That is a great idea!!
 It keeps the api conversion logic out of labels selector.
Done.